sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -169,6 +169,52 @@ def test_f_regression_center():
     assert_almost_equal(F2[0], 0.232558139)  # value from statsmodels OLS
 
 
+@pytest.mark.parametrize(
+    "X, y, expected_f_statistic, expected_p_values",
+    [
+        (
+            # there is a constant feature and the correlation should be 0.0 and
+            # a p-value of 1.0 for this specific feature
+            np.array([[2, 1], [2, 0], [2, 10], [2, 4]]),
+            np.array([0, 1, 1, 0]),
+            np.array([0.0, 0.2293578]),
+            np.array([1.0, 0.67924985]),
+        ),
+        (
+            # the target is constant and thus all f-statistic should be 0.0
+            # and all p-values should be 1.0
+            np.array([[5, 1], [3, 0], [2, 10], [8, 4]]),
+            np.array([0, 0, 0, 0]),
+            np.array([0.0, 0.0]),
+            np.array([1.0, 1.0]),
+        ),
+        (
+            # In this case, we expected an infinite correlation but it should
+            # not raise a warning. The p-values should be 0.0
+            np.array([[0, 1], [1, 0], [2, 10], [3, 4]]),
+            np.array([0, 1, 2, 3]),
+            np.array([np.inf, 0.845433]),
+            np.array([0.0, 0.454913]),
+        ),
+    ],
+)
+def test_f_regression_corner_case(X, y, expected_f_statistic, expected_p_values):
+    """Check that no warnings are raised for some corner cases and that we provide
+    the proper correlation and p-value.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/15672
+    """
+    with pytest.warns(None) as records:
+        f_statistic, p_values = f_regression(X, y)
+    assert len(records) == 0, (
+        f"f_regression raised {len(records)} warnings:"
+        f" {[' '.join(str(w.message) for w in records)]}"
+    )
```

Comment:
I was thinking of moving to this pattern:

```python
assert not [str(w.message) for w in records]
```

for all the places we want to test for non-warnings.